### PR TITLE
iOS SDK Wrapper implementation

### PR DESCRIPTION
**Pull request by marinaserranomontes**
## Pull request checklist

- [ ] All tests pass. Demo project builds and runs.
- [ ] I have resolved any merge conflicts. Confirmation: ____

## What's in this pull request?

>...

### CODE REVIEW

File: ios/OTAcceleratorPackUtilProject/OTAcceleratorPackUtil/OTSDKWrapper.h [hunk 1]
```diff
@@ -55,23 +62,29 @@ typedef void (^OTWrapperBlock)(OTWrapperSignal signal, NSString *streamId, NSErr
  */
 @property (readonly, weak, nonatomic) id<OTSDKWrapperDataSource> dataSource;
 
+@property (weak, nonatomic) id<OTWrapperSignalDelegate> delegate;
```
Comment (southfox) [inline ios/OTAcceleratorPackUtilProject/OTAcceleratorPackUtil/OTSDKWrapper.h]: but not sure

File: ios/OTAcceleratorPackUtilProject/OTAcceleratorPackUtil/OTSDKWrapper.m [hunk 2]
```diff
@@ -2,12 +2,593 @@
...
+        _internalConnectionCount = 0;
+        _connectionsOlderThanMe = 0;
+        _subscribers = [[NSMutableDictionary alloc] init];
+        _streams = [[NSMutableDictionary alloc] init];
+        _connections = [[NSMutableDictionary alloc] init];
+    }
+    return self;
+}
+
+- (NSError *)broadcastSignalWithType:(NSString *)type {
```
Comment (marinaserranomontes) [inline ios/OTAcceleratorPackUtilProject/OTAcceleratorPackUtil/OTSDKWrapper.m]: agree. thanks @southfox

File: ios/OTAcceleratorPackUtilProject/OTAcceleratorPackUtil/OTSDKWrapper.m [hunk 3]
```diff
@@ -2,12 +2,593 @@
...
+        }
+        else {
+            self.handler(OTWrapperDidStopPublishing, _publisher.stream.streamId, error);
+        }
+        _publisher = nil;
+    }
+    
+    //force unsubscriber
+    if ([_subscribers count] != 0) {
+        for(NSString* key in _subscribers) {
```
Comment (marinaserranomontes) [inline ios/OTAcceleratorPackUtilProject/OTAcceleratorPackUtil/OTSDKWrapper.m]: +1

File: ios/OTAcceleratorPackUtilProject/OTAcceleratorPackUtil/OTSDKWrapper.m [hunk 4]
```diff
@@ -2,12 +2,593 @@
...
+        _subscribers = [[NSMutableDictionary alloc] init];
+        _streams = [[NSMutableDictionary alloc] init];
+        _connections = [[NSMutableDictionary alloc] init];
+    }
+    return self;
+}
+
+- (NSError *)broadcastSignalWithType:(NSString *)type {
+    NSError *error;
+    if ( _session ){
```
Comment (marinaserranomontes) [inline ios/OTAcceleratorPackUtilProject/OTAcceleratorPackUtil/OTSDKWrapper.m]: +1

File: ios/OTAcceleratorPackUtilProject/OTAcceleratorPackUtil/OTSDKWrapper.m [hunk 5]
```diff
@@ -2,12 +2,593 @@
...
+    OTError *error = nil;
+    UIView *view = nil;
+    
+    if (!self.publisher){
+        //create a new publisher
+        self.publisher = [[OTPublisher alloc] initWithDelegate:self name:self.name];
+    }
+    //start publishing
+    [self.session publish:self.publisher error:&error];
+    view = self.publisher.view;
```
Comment (marinaserranomontes) [inline ios/OTAcceleratorPackUtilProject/OTAcceleratorPackUtil/OTSDKWrapper.m]: +1

File: ios/OTAcceleratorPackUtilProject/OTAcceleratorPackUtil/OTSDKWrapper.m [hunk 6]
```diff
@@ -2,12 +2,593 @@
...
+    if (error) {
+        self.handler(OTWrapperDidFail, nil, error);
+    }
+    
+    return view;
+}
+
+- (NSError *)stopPublishingLocalMedia {
+    OTError *error = nil;
+    if ( self.publisher ) {
```
Comment (marinaserranomontes) [inline ios/OTAcceleratorPackUtilProject/OTAcceleratorPackUtil/OTSDKWrapper.m]: +1

File: ios/OTAcceleratorPackUtilProject/OTAcceleratorPackUtil/OTSDKWrapper.m [hunk 7]
```diff
@@ -2,12 +2,593 @@
...
+    
+    [_connections removeObjectForKey:connection.connectionId];
+    
+    //check creationtime of the connections
+    [self compareConnectionTimeWithConnection: connection];
+}
+
+- (void)session:(OTSession *)session streamCreated:(OTStream *)stream {
+    if( !_streams[stream.streamId]){
+        [_streams setObject:stream forKey:stream.streamId];
```
Comment (marinaserranomontes) [inline ios/OTAcceleratorPackUtilProject/OTAcceleratorPackUtil/OTSDKWrapper.m]: +1

File: ios/OTAcceleratorPackUtilProject/OTAcceleratorPackUtil/OTSDKWrapper.m [hunk 8]
```diff
@@ -2,12 +2,593 @@
...
+#pragma mark - session
+- (instancetype)initWithDataSource:(id<OTSDKWrapperDataSource>)dataSource {
+    
+    return [self initWithName:[NSString stringWithFormat:@"%@-%@", [UIDevice currentDevice].systemName, [UIDevice currentDevice].name]
+                   dataSource:dataSource];
+}
+
+- (instancetype)initWithName:(NSString *)name
+                  dataSource:(id<OTSDKWrapperDataSource>)dataSource {
+    if (!dataSource) {
```
Comment (marinaserranomontes) [inline ios/OTAcceleratorPackUtilProject/OTAcceleratorPackUtil/OTSDKWrapper.m]: I have not a strong opinion in this case. I think we are using the if condition in other cases, but NSAssert(.....,@"...."); could be working as well. @Lucashuang0802 wdyt?

File: ios/OTAcceleratorPackUtilProject/OTAcceleratorPackUtil/OTSDKWrapper.h (hunk 1 above)
Comment (Lucashuang0802) [inline ios/OTAcceleratorPackUtilProject/OTAcceleratorPackUtil/OTSDKWrapper.h]: `weak` directive will break it automatically tho.

Comment (southfox) [inline ios/OTAcceleratorPackUtilProject/OTAcceleratorPackUtil/OTSDKWrapper.h]: yes, looks fine.

**Pull request merged**